croncat/src/channels.rs

```diff
@@ -18,7 +18,11 @@ pub type BlockStreamRx = async_broadcast::Receiver<tendermint::Block>;
 /// Create a block stream channel of a specified size, used to create back pressure.
 ///
 pub fn create_block_stream(channel_size: usize) -> (BlockStreamTx, BlockStreamRx) {
-    async_broadcast::broadcast(channel_size)
+    let mut channel = async_broadcast::broadcast(channel_size);
+    // TODO: THIS IS A HACK. Something is not flushing the channel properly.
+    // For now all our tasks should be received fine.
+    channel.0.set_overflow(true);
```

Comment:
Great solution!

Reply:
BTW don't think it is a `hack`, don't think we want to use any RAM to process old blocks and we don't want it to crash in case of queue overflow or even worse OOM

Reply:
It is kind of a hack, what was happening is that the broadcast channel would fill up even with all the listeners using the messages. This was causing a deadlock and it's what was making the agent "crash" when the channel buffer was full.

Reply:
When overflow is set to true, it will flush by removing old messages right?

Reply:
Yup.